### ISSUE TITLE
Set global_context_cache._platform use CUDA and mixed precision

### ISSUE THREAD

**Issue by zhang-ivy** [choderalab/perses#901]
When running a perses h-repex simulation with 60 replicas on lilac (using 1 node), I was seeing that on lx nodes, the job was running out of memory and subsequently using the CPU without throwing any warnings. 

I added loggers to print out the platform. Here is a snippet of the log for the first iteration:
```
DEBUG:openmmtools.utils:LangevinSplittingDynamicsMove: Context request took    5.055s
DEBUG:openmmtools.utils:LangevinSplittingDynamicsMove: step(250) took  729.812s
DEBUG:openmmtools.utils:LangevinSplittingDynamicsMove: update sampler state took    0.002s
DEBUG:openmmtools.mcmc:LangevinSplittingDynamicsMove: Context obtained, platform is CPU
DEBUG:openmmtools.utils:LangevinSplittingDynamicsMove: Context request took    5.136s
DEBUG:openmmtools.utils:LangevinSplittingDynamicsMove: step(250) took  772.275s
DEBUG:openmmtools.utils:LangevinSplittingDynamicsMove: update sampler state took    0.001s
DEBUG:openmmtools.mcmc:LangevinSplittingDynamicsMove: Context obtained, platform is CPU
DEBUG:openmmtools.utils:LangevinSplittingDynamicsMove: Context request took    5.172s
DEBUG:openmmtools.utils:LangevinSplittingDynamicsMove: step(250) took  704.387s
DEBUG:openmmtools.utils:LangevinSplittingDynamicsMove: update sampler state took    0.001s
DEBUG:openmmtools.mcmc:LangevinSplittingDynamicsMove: Context obtained, platform is CPU
DEBUG:openmmtools.utils:LangevinSplittingDynamicsMove: Context request took    4.738s
DEBUG:openmmtools.utils:LangevinSplittingDynamicsMove: step(250) took  683.204s
DEBUG:openmmtools.utils:LangevinSplittingDynamicsMove: update sampler state took    0.002s
DEBUG:openmmtools.mcmc:LangevinSplittingDynamicsMove: Context obtained, platform is CPU
DEBUG:openmmtools.utils:LangevinSplittingDynamicsMove: Context request took    4.859s
DEBUG:openmmtools.utils:LangevinSplittingDynamicsMove: step(250) took  711.483s
DEBUG:openmmtools.utils:LangevinSplittingDynamicsMove: update sampler state took    0.002s
DEBUG:mpiplus.mpiplus:Running _get_replica_move_statistics serially.
DEBUG:openmmtools.utils:Propagating all replicas took 11295.271s
``` 

We should set global_context_cache._platform use CUDA and mixed precision in the future, so that an error will be raised if no CUDA context is available.

**Comment by zhang-ivy** [comment]
I think this has been addressed by @ijpulidos's context cache PR, so this can be closed, right?

**Comment by jchodera** [comment]
Fixed by #938
